GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -198,28 +201,43 @@ private bool TryRunCleanUp(out string error)
         {
             error = null;
 
-            string errorMessage = null;
+            string errorMessage = string.Empty;
             if (!this.LaunchInsideSpinner(
                 () =>
                 {
-                    if (!this.TryReleaseUpgradeLock(out errorMessage))
+                    bool success = true;
+                    string unlockError;
+                    if (!this.TryReleaseUpgradeLock(out unlockError))
                     {
-                        return false;
+                        errorMessage += unlockError + Environment.NewLine;
+                        success = false;
                     }
 
-                    if (this.remount && !this.preRunChecker.TryMountAllGVFSRepos(out errorMessage))
+                    string remountError;
+                    if (this.remount && !this.preRunChecker.TryMountAllGVFSRepos(out remountError))
                     {
                         EventMetadata metadata = new EventMetadata();
                         metadata.Add("Upgrade Step", nameof(this.TryRunCleanUp));
-                        this.tracer.RelatedError(metadata, $"{nameof(this.preRunChecker.TryMountAllGVFSRepos)} failed. {errorMessage}");
-                        return false;
+                        this.tracer.RelatedError(metadata, $"{nameof(this.preRunChecker.TryMountAllGVFSRepos)} failed. {remountError}");
+                        errorMessage += remountError + Environment.NewLine;
+                        success = false;
                     }
 
-                    return true;
+                    string downloadsCleanupError;
+                    if (this.deleteDownloadedAssets && !this.upgrader.TryCleanup(out downloadsCleanupError))

```

Comment:
Is there anything that will come back through and delete these files if this step fails? What is the actual goal here? Is it just a best-effort cleanup or do we care about making it reliable?

Reply:
- GVFS Un-installer removes downloaded assets that are left behind, but otherwise the assets will remain there when delete fails. 
- The goal is to try to throw away downloaded assets after use, so they don't pile up.
- Reliability - We don't want unwanted disk usage. Upgrader tool runs elevated, so I'm not expecting this delete to fail. The case where user terminates the tool before it gets to do the cleanup is not handled now.